File: Atividades A1/15 Questões/Java/Questoes15/src/Questao12/Main.java
package Questao12;
//12. Sobrecarga de Operadores (Python) / Métodos de Conveniência (Java, Go) Em Python, sobrecarregue o operador + para somar dois objetos Produto baseados no preço.
//Em Java e Golang, crie métodos que permitam essa funcionalidade.

class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public double getPreco() {
        return preco;
    }

    public Produto somar(Produto outroProduto) {
        double soma = this.preco + outroProduto.getPreco();
        return new Produto("Pacote com " + this.nome + " e " + outroProduto.nome, soma);
    }

    @Override
    public String toString() {
        return nome + " - R$" + preco;
    }
}

public class Main {
    public static void main(String[] args) {
        Produto iPhone = new Produto("iPhone", 6000.00);
        Produto airpods = new Produto("Airpods", 1500.00);

        Produto pacote = iPhone.somar(airpods);

        System.out.println(pacote);
    }
}

File: Atividades A1/15 Questões/Python/Questoes15/Questao12/Main.py
#12. Sobrecarga de Operadores (Python) / Métodos de Conveniência (Java, Go) Em Python, sobrecarregue o operador + para somar dois objetos Produto baseados no preço.
#Em Java e Golang, crie métodos que permitam essa funcionalidade.

class Produto:
    def __init__(self, nome, preco):
        self.nome = nome
        self.preco = preco

    def __add__(self, outroProduto):
        soma = self.preco + outroProduto.preco
        novoNome = f'Pacote com {self.nome} e {outroProduto.nome}'
        return Produto(novoNome, soma)

    def __str__(self):
        return f'{self.nome} - R${self.preco:.2f}'


if __name__ == '__main__':
    iphone = Produto('iPhone', 6000.00)
    airpods = Produto('Airpods', 1500.00)

    pacote = iphone + airpods

    print(pacote)